lightning/src/routing/gossip.rs

```diff
@@ -130,6 +134,18 @@ pub struct NetworkGraph<L: Deref> where L::Target: Logger {
 	// Lock order: channels -> nodes
 	channels: RwLock<BTreeMap<u64, ChannelInfo>>,
 	nodes: RwLock<BTreeMap<NodeId, NodeInfo>>,
+	#[cfg(feature = "std")]
+	/// Keeps track of [`NodeId`]s we have explicitly removed due to permanent failure so that we
+	/// don't resync them from gossip. Each [`NodeId`] is mapped to the instant when we removed it
+	/// from the network graph so that we can forget we removed it after some time (by removing it
+	/// from this map) and it can be resynced from gossip if it appears again.
+	removed_nodes: Mutex<HashMap<NodeId, Instant>>,
+	#[cfg(feature = "std")]
+	/// Keeps track of short channel IDs for channels we have explicitly removed due to permanent
+	/// failure so that we don't resync them from gossip. Each SCID is mapped to the instant when we
+	/// removed it from the network graph so that we can forget we removed it after some time (by
+	/// removing it from this map) and it can be resynced from gossip if it appears again.
+	removed_channels: Mutex<HashMap<u64, Instant>>,
```

Comment:
Could you take a look at [this](https://github.com/lightningdevkit/rust-lightning/pull/1649#discussion_r966784103)?

I will ack once we come to a conclusion there...